mmeval/metrics/bleu.py

```diff
@@ -105,21 +78,56 @@ def __init__(self,
             ngram_weights = [1.0 / n_gram] * n_gram
         self.ngram_weights = ngram_weights
 
-    def add(self, predictions: Sequence[str], references: Sequence[Sequence[str]]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        # Select tokenizer according to the entered value.
+        self.tokenizer_fn = None
+        if callable(tokenizer_fn):
+            self.tokenizer_fn = tokenizer_fn
+        elif isinstance(tokenizer_fn, str):
+            self.tokenizer_fn = get_tokenizer(tokenizer_fn)
+            if self.tokenizer_fn is None:
+                raise ValueError(
+                    "Right now, `tokenizer_fn` only supports pre-defined 'en' or 'cn'."  # noqa: E501
+                )
+        else:
+            assert tokenizer_fn is None, \
+                f'`tokenizer_fn` supports Callable, str or None, but not `{type(tokenizer_fn)}`'  # noqa: E501
+
+    def add(self, predictions: Union[str, Sequence[str]], references: Union[str, Sequence[str], Sequence[Sequence[str]]]) -> None:  # type: ignore # yapf: disable # noqa: E501
         """Add the intermediate results to ``self._results``.
 
         Args:
-             predictions (Sequence[str]): An iterable of machine
-                translated corpus.
-             references (Sequence[Sequence[str]]): An iterable of
-                iterables of reference corpus.
+             predictions (Union[Sequence[str] | str]): An iterable of
+                predicted sentences or a single predicted sentence.
+             references (Union[Sequence[str] | str]): An iterable of
+                referenced sentences or an iterable of target sentences
+                or a single target sentence.
         """
-
+        if isinstance(references, list) and all(
+                isinstance(reference, str) for reference in references):
+            if isinstance(predictions, str):
+                references = [references]
+            else:
+                if len(predictions) == 1:
+                    references = [references]
+                else:
+                    references = [[reference] for reference in references]
+
+        if isinstance(predictions, str):
+            predictions = [predictions]
+
+        if isinstance(references, str):
```

Comment:
If references is sequence[str], should it be wrapped with []?

Reply:
In lines 105-113 of the code, the situation of sequence [str] is analyzed